src/lib.rs

```diff
@@ -168,9 +166,21 @@ pub fn run_file(mut config: Config, path: &Path) -> Result<std::process::ExitSta
     Ok(build_command(path, &config, "", &comments).status()?)
 }
 
-pub fn run_tests_generic(config: Config, file_filter: impl Fn(&Path) -> bool + Sync) -> Result<()> {
-    // Get the triple with which to run the tests
-    let target = config.target();
+pub fn run_tests_generic(
+    mut config: Config,
+    file_filter: impl Fn(&Path) -> bool + Sync,
+) -> Result<()> {
+    // Make sure we have the host and target triples.
+    if config.host.is_none() {
+        config.host = Some(
+            rustc_version::VersionMeta::for_command(std::process::Command::new(&config.program))
+                .expect("failed to parse rustc version info")
+                .host,
+        );
+    }
+    if config.target.is_none() {
+        config.target = Some(config.host.clone().unwrap());
+    }
```

Comment:
I think this should be a method on `Config` to allow easier re-use in the code paths that don't use `run_tests`

Reply:
Fair, done.